test/integration/redirects_test.rb

```diff
@@ -142,4 +162,197 @@ def test_show_past_glossary_term
+
+  # merge_herbaria (get)          Herbaria::Merges#new (get)
+  def test_merge_herbaria
+    fundis = herbaria(:fundis_herbarium)
+    assert_true(fundis.owns_all_records?(mary))
+    marys = mary.create_personal_herbarium
+    login("mary")
+
+    get("/herbarium/merge_herbaria?this=#{fundis.id}&that=#{marys.id}")
+    # fundis ends up being the destination because it is older.
+    assert_equal(herbaria_path(id: fundis.id), @response.request.fullpath)
+  end
+
+  # next_herbarium (get)          herbaria::Nexts#show { next: "next" } (get)
+  def test_next_herbarium
+    query = Query.lookup_and_save(:Herbarium, :all)
+    assert_operator(query.num_results, :>, 1,
+                    "Test needs query with > 1 result")
+    number1 = query.results[0]
+    number2 = query.results[1]
+
+    assert_old_url_redirects_to_new_path(
+      :get,
+      "/herbarium/next_herbarium/#{number1.id}",
+      herbarium_path(number2, q: query.record.id.alphabetize)
+    )
+  end
+
+  # prev_herbarium (get)          herbaria::Nexts#show { next: "prev" } (get)
+  def test_prev_herbarium
+    query = Query.lookup_and_save(:Herbarium, :all)
+    assert_operator(query.num_results, :>, 1,
+                    "Test needs query with > 1 result")
+
+    number1 = query.results[0]
+    number2 = query.results[1]
+
+    assert_old_url_redirects_to_new_path(
+      :get,
+      "/herbarium/prev_herbarium/#{number2.id}",
+      herbarium_path(number1, q: query.record.id.alphabetize)
+    )
+  end
+
+  # request_to_be_curator (get)   Herbaria::CuratorRequest#new (get)
+  def test_request_to_be_herbarium_curator_get
+    nybg = herbaria(:nybg_herbarium)
+    login("mary")
+
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/request_to_be_curator/#{nybg.id}",
+      new_herbaria_curator_request_path(id: nybg)
+    )
+  end
+
+  # request_to_be_curator (post)  Herbaria::CuratorRequest#create (post)
+  def test_request_to_be_herbarium_curator_post
+    nybg = herbaria(:nybg_herbarium)
+    email_count = ActionMailer::Base.deliveries.count
+    login("mary")
+    post("/herbarium/request_to_be_curator/#{nybg.id}")
+
+    # Rails seems to follow the redirect, instead of just redirecting
+    assert_equal(herbarium_path(nybg), @response.request.fullpath)
+    assert_equal(email_count + 1, ActionMailer::Base.deliveries.count)
+  end
+
+  # show_herbarium (get)          show (get)
+  def test_show_herbarium_get
+    nybg = herbaria(:nybg_herbarium)
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/show_herbarium/#{nybg.id}", herbarium_path(nybg)
+    )
+  end
+
+  # show_herbarium (post)         Herbaria::Curators#create (post)
+  def test_show_herbarium_post
+    nybg = herbaria(:nybg_herbarium)
+    assert(nybg.curators.include?(rolf))
+    curator_count = nybg.curators.count
+    login("rolf")
+    post("/herbarium/show_herbarium?id=#{nybg.id}&add_curator=#{mary.login}")
+
+    assert_equal(herbarium_path(nybg), @response.request.fullpath)
```

Comment:
I don't understand why the response is :success rather than a 301. 
Obviously a consequence of my routing. 
How can I fix this?